src/Microsoft.AspNetCore.Routing/Matching/CandidateSet.cs

```diff
@@ -46,18 +75,18 @@ public CandidateSet(Endpoint[] endpoints, int[] scores)
                     return;
 
                 case 1:
-                    _state0 = new CandidateState(endpoints[0], score: scores[0]);
+                    _state0 = new CandidateState(endpoints[0], values[0], scores[0]);
```

Comment:
nice to have: a test that verifies values was used

Reply:
Yeah, you're right.